src/main/cpp/util/md5.cc

```diff
@@ -338,7 +340,11 @@ void Md5Digest::Transform(
 
 string Md5Digest::String() const {
   string result;
-  b2a_hex(reinterpret_cast<const uint8_t*>(state), &result, 16);
+  unsigned int state_le[4];
```

Comment:
And this one concatenates the state represented as four 32-bit low-endian numbers into a bit string, then returns the result a  string of hex digits. IOW, the byte order of `state_le` should be the same on every architecture.

Reply:
Yes. the same after the flip.
- When we enter the function state is 4 32-bit numbers in host order
- Then we make sure state_le is 4 numbers in little endian order
- Then we hexify that.

I could rewrite this with something portable, that would be more like `words_to_hex(state, result, 4)`
But... more reviewer complexity.